api/src/task/opportunities/store_opportunity_version_task.py

```diff
@@ -26,11 +33,26 @@ def run_task(self) -> None:
         latest_time = latest_job.created_at if latest_job else datetime(1970, 1, 1)
 
         opportunity_change_audits = self.db_session.scalars(
-            select(OpportunityChangeAudit).where(OpportunityChangeAudit.updated_at > latest_time)
+            select(OpportunityChangeAudit)
+            .where(OpportunityChangeAudit.updated_at > latest_time)
+            .options(selectinload("*"))
         ).all()
 
         for opp_change_audit in opportunity_change_audits:
+            log_extra = {
+                "opportunity_id": opp_change_audit.opportunity.opportunity_id,
+            }
```

Comment:
Not necessary, but you can also do something like this:
```suggestion
            log_extra = {
                "opportunity_id": opp_change_audit.opportunity_id,
            }
```